src/gmt_api.c

```diff
@@ -10404,6 +10404,15 @@ struct GMT_RESOURCE *GMT_Encode_Options (void *V_API, const char *module_name, i
 		}
 		deactivate_output = true;	/* Turn off implicit table output since only secondary -G -G -G is in effect */
 	}
+	/* 1l. Check if this is makecpt using -E or -S with no args */
+	else if (!strncmp (module, "makecpt", 7U) && ((opt = GMT_Find_Option (API, 'E', *head)) || (opt = GMT_Find_Option (API, 'S', *head))) && opt->arg[0] == '\0') {
+		/* Found the -E or -S option without arguments */
+		gmt_M_str_free (opt->arg);
```

Comment:
Ok, tried it and like I suspected. The program now hangs.

Reply:
I built and tried it in Matlab and all my cases worked.  I tried -E, -E10, -S, -Sr and all worked in matlab for me.  The changes are only in gmt_api.c and makecpt.c so not sure why this is not working in Julia.

Reply:
No Matlab either. When I do

```
gmtmex('makecpt -E0 -Crainbow', rand(10,1))
```

it crashes at makecpt line 468
```
			case DO_ROUNDED_RANGE_E:	/* -E only */
				Ctrl->T.T.inc = (D->max[zcol] - D->min[zcol]) / (Ctrl->E.levels - 1);
```
because the D struct has no data (``D->max`` is invalid mem)

Reply:
I will try again.  I admit I passed a file in matlab and not a memory item.

Reply:
OK, agree that it fails.  Need to regroup on this one.

Reply:
Yes, that's what I called an optional primary input. No, no other module with this option come to my mind, but there are soooooooo many options.

Reply:
OK, I will try to find time on the weekend but that weekend is packed with work already.

Reply:
No hurry. Meanwhile I've implemented a [workaround ](https://github.com/GenericMappingTools/GMT.jl/blob/master/src/psxy.jl#L321)

Reply:
Please accept this PR; it does a few things related to the makecpt stuff but mostly it adds UTF8 support in postscriptlight.  AFter update, see and try the utf8.sh test in pstext.

Reply:
Sorry, I'm confused. I see no further changes in this PR, namely no utf8.sh file.